Description: Install MokManager if present
 If MokManager is present on the host system, copy it onto the EFI boot
 partition for use.
Author: Steve Langasek <steve.langasek@ubuntu.com>
Forwarded: no
Last-Update: 2013-09-23

Index: b/util/grub-install.in
===================================================================
--- a/util/grub-install.in
+++ b/util/grub-install.in
@@ -848,6 +848,12 @@
 	    fi
 	    cp "$shim_signed" "${efidir}/${efi_file}"
 	    cp "$efi_signed" "${efidir}/grub$efi_suffix.efi"
+	    # Not critical, so not an error if it's not present (as it won't
+	    # be for older releases); but if we have it, make sure it's
+	    # installed.
+	    if [ -e /usr/lib/shim/MokManager.efi.signed ]; then
+		cp /usr/lib/shim/MokManager.efi.signed "${efidir}/MokManager.efi"
+	    fi
 	else
 	    cp "$efi_signed" "${efidir}/${efi_file}"
 	fi
